scale/workload.py

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/python3
+
+"""Provide functions to analyze current workload status of the cluster.
+
+All functions in the file should be read-only and cause no side effects."""
+
+from utils import get_nodes, is_unschedulable, get_pods, get_pod_namespace, \
+    get_pod_type, get_pod_host_name, get_name
+from settings import CAPACITY_PER_NODE, MIN_NODES, MAX_NODES, MAX_UTILIZATION, MIN_UTILIZATION, OPTIMAL_UTILIZATION
+from settings import OMIT_NAMESPACES, CRITICAL_POD_TYPES, OMIT_POD_TYPES, CRITICAL_NAMESPACES
+import logging
+scale_logger = logging.getLogger("scale")
+
+
+def get_pods_number_on_node(node, pods=get_pods()):
```

Comment:
setting the default to a dynamic value (such as returned by a function) will usually not behave as expected. As an example, if you run:

```python
import time
from datetime import datetime

def p(w=datetime.now()):
    print(w)

p()
time.sleep(10)
p()
```

You might expect that two calls to p will print two datetimes that are ten seconds apart. However, what it actually prints is:

```
2017-02-20 22:28:48.010126
2017-02-20 22:28:48.010126
```

So the value of the parameter is the output of the function when it is run for the first time, *when the defining function code is parsed*, not when called. You almost never want this :) 

You can either let pods be a required param, or just make an api query inside the function